src/graph.rs

```diff
@@ -1308,29 +1456,111 @@ mod tests {
         assert_eq!(closeness, expected_closeness);
     }
 
-    // Helper function to create an AGraph from a json file.
+    // Helper function to create a sample from a json file.
     // The file will begin like this:
-    //   {"agraph":[[2630,3217,1608,1035,...
+    //   {"indices":[[2630,3217,1608,1035,...
     // and end like this:
     //   ...2316,1068,1238,704,2013]]}
-    pub fn load_agraph(filepath: &str) -> AGraph {
+    pub fn load_sample(filepath: &str) -> Sample {
         let jstring = fs::read_to_string(filepath).unwrap();
-        let agraph_sample: AGraphSample = serde_json::from_str(&jstring).unwrap();
-        agraph_sample.agraph
+        let sample: Sample = serde_json::from_str(&jstring).unwrap();
+        sample
     }
 
     #[test]
     #[ignore = "takes a while to run"]
-    fn closeness_imported_sample_3226() {
-        let agraph = load_agraph("testdata/agraph-3226.json");
-        assert_eq!(agraph.len(), 3226);
-        let graph: Graph<usize> = Graph::new();
-        let start = Instant::now();
-        let (betweenness, closeness) = graph.compute_betweenness_and_closeness(&agraph);
-        let elapsed = start.elapsed();
-        println!("elapsed for 3226 nodes: {elapsed:?}");
-        assert!(elapsed.as_secs() < 45);
-        assert_eq!(agraph.len(), betweenness.len());
-        assert_eq!(agraph.len(), closeness.len());
+    fn loaded_sample_graph() {
+        let sample = load_sample("testdata/sample.json");
+
+        // graph 1 uses integers as node value
+        let mut graph1 = Graph::new();
+        let mut n = 0;
+        for node in &sample.indices {
+            for connection in node {
+                if *connection > n {
+                    graph1.insert(Edge::new(n, *connection));
+                }
+            }
+            n += 1;
+        }
+
+        let betweenness_centrality1 = graph1.betweenness_centrality();
+        let closeness_centrality1 = graph1.closeness_centrality();
+
+        // graph2 uses ip address as node value
+        let mut graph2: Graph<&str> = Graph::new();
+        let mut n = 0;
+        for node in &sample.indices {
+            for connection in node {
+                if *connection > n {
+                    graph2.insert(Edge::new(
+                        &sample.node_ips[n],
+                        &sample.node_ips[*connection],
+                    ));
+                }
+            }
+            n += 1;
+        }
+
+        let betweenness_centrality2 = graph2.betweenness_centrality();
+        let closeness_centrality2 = graph2.closeness_centrality();
+        let b1 = betweenness_centrality1.get(&0).unwrap();
+        let b2 = betweenness_centrality2.get("65.21.141.242").unwrap();
+        let c1 = closeness_centrality1.get(&0).unwrap();
+        let c2 = closeness_centrality2.get("65.21.141.242").unwrap();
+        assert_eq!(b1, b2);
+        assert_eq!(c1, c2);
+
+        // Tobolsk (index 1837) has betweenness 9.576638518159478e-8
```

Comment:
```suggestion
        // Index 1837 has betweenness 9.576638518159478e-8
```
Since there is no notion of location in the test data.

Reply:
done